docs/lesson-plans/week-1.md

```diff
@@ -0,0 +1,649 @@
+end up deleting your entire computer (there are a couple protections, but would you really want
+to test them out?).
+
+```shell script
+# Delete the test.txt file
+rm test.txt
+
+# Delete the test directory
+rm -rf test
+```
+
+#### cat
+
+`cat` prints out the contents of a file. It technically stands for 'concatenate' file to 
+standard output (your console).
+
+Format:
+```shell script
+cat <filepath>
+```
+
+```shell script
+# Print out contents of test.txt
+cat test.txt
+```
+
+#### vim
+
+`vim` is a text editor that usually comes preinstalled on most terminals. It allows you
+to open files, edit them using quick key binds, and write/close the files. It's notorious
+for having a really steep learning curve, so people tend to avoid it, but it can be a 
+powerful tool if you do end up spending time trying to learn it.
+
+![text-editor-learning-curves](./img/text-editor-learning-curves.png)
+
+We're showing this since you'll have to at some point learn a terminal-based text editor.
+It's also faster sometimes for quick edits to a file.
+
+Before you even try to open it, though, you need to know how to exit it. The easiest way to
+do that is to type <esc>, then ":q!".
+
+Format:
+```shell script
+# Open a file using vim
+vim <filepath>
+
+# vimtutor (see below)
+vimtutor
+```
+
+- You start in 'normal' mode
+- To edit a file, type 'i' to go into 'insert' mode.
+- To get back into normal mode, type <esc>
+- To save a file, type ":w" in normal mode
+- (again) To exit, type ":q!" in normal mode
+- The "q!" means quit without saving, so you can also do ":wq" to write and then quit
+- Try out the `vimtutor` command if you want to learn how to actually use vim.
+
+#### nano
+
+`nano` is another terminal-based text editor. It's easier to use compared to vim and it has the 
+controls on the bottom of the screen, but it's (in Conner's opinion) a horrible experience 
+that leaves him annoyed every time he uses it. If you're dedicated to learning a really fast
+text editor, learn vim, but I'm leaving this here for people who don't want to.
+
+Example:
+```shell script
+nano <filepath>
+```
+
+#### sudo
+
+`sudo` stand for "super-user do", and it's required at times where elevated permissions are required.
+This may be when you're dealing with system configurations or programs that need access to possibly 
+sensitive information, so your password will be required to allow it to make changes. This is sometimes a good
+thing, but you should definitely not just start using it whenever a program isn't working in a way
+you want it to. Again, remember that this gives permission to whatever program you're calling it with,
+so you should be a little careful about using it. You can read a little bit more about that 
+[here](https://www.linux.com/news/sudo-or-not-sudo-question/).
+
+Format:
+```shell script
+sudo <some other command>
+```
+
+- An example of a time where it may be useful to use it is when globally installing an npm package. 
```

Comment:
I still don't understand if "a" should be used for letters that have vowels in their beginning or "an" 
a npm package or an npm package?

Reply:
I'm going to go off of [this](https://blog.apastyle.org/apastyle/2012/04/using-a-or-an-with-acronyms-and-abbreviations.html), which says it should be 'an npm'. Thoughts?

Reply:
sounds goood :)